src/Raven.Server/Documents/TcpHandlers/TcpConnectionOptions.cs

```diff
@@ -93,6 +98,13 @@ public void Dispose()
             // a possible race condition on dispose
         }
 
+#if !RELEASE
+        ~TcpConnectionOptions()
+        {
+            throw new LowMemoryException($"Detected a leak on TcpConnectionOptions ('{ToString()}') when running the finalizer.");
```

Comment:
Not sure it is a good idea to throw in the finalizer... if this for test running, better is simply to write to console.

Reply:
This is only in `!RELEASE`, we are already doing this in HttpCache.

Reply:
And no one read the console logs